nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessAnalysis.java

```diff
@@ -211,12 +212,25 @@ public NullnessStore getNullnessInfoBeforeNestedMethodNode(
     return store.filterAccessPaths(
         (ap) -> {
           boolean allAPNonRootElementsAreFinalFields = true;
-          for (AccessPathElement ape : ap.getElements()) {
+          ImmutableList<AccessPathElement> elements = ap.getElements();
+          for (int i = 0; i < elements.size(); i++) {
+            AccessPathElement ape = elements.get(i);
             Element e = ape.getJavaElement();
-            if (!e.getKind().equals(ElementKind.FIELD)
-                || !e.getModifiers().contains(Modifier.FINAL)) {
-              allAPNonRootElementsAreFinalFields = false;
-              break;
+            if (i != elements.size() - 1) { // "inner" elements of the access path, must be fields
+              if (!e.getModifiers().contains(Modifier.FINAL)) {
+                allAPNonRootElementsAreFinalFields = false;
+                break;
+              }
+            } else { // last element; can be a final field or @MonotonicNonNull field
+              if (!e.getModifiers().contains(Modifier.FINAL)
+                  && !e.getAnnotationMirrors().stream()
+                      .anyMatch(
+                          am ->
+                              Nullness.isMonotonicNonNullAnnotation(
+                                  am.getAnnotationType().toString()))) {
+                allAPNonRootElementsAreFinalFields = false;
+                break;
```

Comment:
nit: unnecessary break since this is the last iteration.

Reply:
70f854226bc2e8f8d5812f6211f12fe47f03a295